cmd/packages.go

```diff
@@ -236,54 +235,66 @@ func isVerbose() (result bool) {
 	return appConfig.CliOptions.Verbosity > 0 || isPipedInput
 }
 
+func generateSBOM(userInput string, errs chan error) (*sbom.SBOM, *source.Source, error) {
+	tasks, err := tasks()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	src, cleanup, err := source.New(userInput, appConfig.Registry.ToOptions(), appConfig.Exclusions)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to construct source from user input %q: %w", userInput, err)
+	}
+	if cleanup != nil {
+		defer cleanup()
+	}
+
+	s := sbom.SBOM{
+		Source: src.Metadata,
+		Descriptor: sbom.Descriptor{
+			Name:          internal.ApplicationName,
+			Version:       version.FromBuild().Version,
+			Configuration: appConfig,
+		},
+	}
+
+	s = buildRelationships(s, src, tasks, errs)
+
+	return &s, src, nil
+}
+
+func buildRelationships(s sbom.SBOM, src *source.Source, tasks []task, errs chan error) sbom.SBOM {
```

Comment:
instead of passing a copy of `sbom.SBOM` and returning `sbom.SBOM` it looks like this is a good candidate for passing a reference to the sbom and returning nothing

Reply:
updated to pass by pointer with no return